src/test/java/com/vinylteam/vinyl/web/servlets/ShopServletTest.java

```diff
@@ -0,0 +1,123 @@
+package com.vinylteam.vinyl.web.servlets;
+
+import com.vinylteam.vinyl.entity.Role;
+import com.vinylteam.vinyl.entity.Shop;
+import com.vinylteam.vinyl.entity.User;
+import com.vinylteam.vinyl.service.ShopService;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InOrder;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.*;
+
+class ShopServletTest {
+
+    private final ShopService mockedShopService = mock(ShopService.class);
+    private final ShopServlet shopServlet = new ShopServlet(mockedShopService);
+
+    private final HttpServletRequest mockedRequest = mock(HttpServletRequest.class);
+    private final HttpServletResponse mockedResponse = mock(HttpServletResponse.class);
+    private final HttpSession mockedHttpSession = mock(HttpSession.class);
+    private final User mockedUser = mock(User.class);
+    private final PrintWriter mockedPrintWriter = mock(PrintWriter.class);
+    private final InOrder inOrderResponse = inOrder(mockedResponse);
+    private final InOrder inOrderRequest = inOrder(mockedRequest);
+    private final List<Shop> testListOfShops;
+
+    ShopServletTest(){
+        testListOfShops = new ArrayList<>();
+        for (int i = 1; i < 4; i++) {
+            Shop shop = new Shop();
+            shop.setId(i);
+            shop.setName("shop" + i);
+            shop.setMainPageLink(shop.getName() + "/main");
+            shop.setImageLink(shop.getName() + "/image.png");
+            shop.setSmallImageLink(shop.getName() + "/smallImage.png");
+            testListOfShops.add(shop);
```

Comment:
You should change method DataGeneratorFroTests.getShopsList(): make list generate 3 elements, instead of 2, add setsmallImageLink(), - and use it.